plugins/network-elements/nuage-vsp/src/com/cloud/network/manager/NuageVspManagerImpl.java

```diff
@@ -285,60 +507,341 @@ public boolean deleteNuageVspDevice(DeleteNuageVspDeviceCmd cmd) {
+                DomainVO domain = (DomainVO) args;
+                try {
+                    List<NuageVspDeviceVO> nuageVspDevices = _nuageVspDao.listAll();
+                    for (NuageVspDeviceVO nuageVspDevice : nuageVspDevices) {
+                        HostVO host = findNuageVspHost(nuageVspDevice.getHostId());
+                        SyncDomainCommand cmd = new SyncDomainCommand(domain.getUuid(), domain.getName(), domain.getPath(), false, true);
+                        _agentMgr.easySend(host.getId(), cmd);
+                    }
+                } catch (Exception e) {
+                    s_logger.error(e.getMessage());
+                }
+            }
+        });
+    }
+
+    @DB
+    private void initNuageVspResourceListeners() {
+        _agentMgr.registerForHostEvents(new Listener() {
+            @Override
+            public boolean processAnswers(long agentId, long seq, Answer[] answers) {
+                return true;
+            }
+
+            @Override
+            public boolean processCommands(long agentId, long seq, Command[] commands) {
+                if (commands != null && commands.length == 1) {
+                    Command command = commands[0];
+                    if (command instanceof PingNuageVspCommand) {
+                        PingNuageVspCommand pingNuageVspCommand = (PingNuageVspCommand) command;
+                        if (pingNuageVspCommand.shouldAudit()) {
+                            Host host = _hostDao.findById(pingNuageVspCommand.getHostId());
+                            auditHost((HostVO) host);
+                        }
+                    }
+                }
+                return true;
+            }
+
+            @Override
+            public AgentControlAnswer processControlCommand(long agentId, AgentControlCommand cmd) {
+                return null;
+            }
+
+            @Override
+            public void processConnect(Host host, StartupCommand cmd, boolean forRebalance) throws ConnectionException {
+
+            }
+
+            @Override
+            public boolean processDisconnect(long agentId, Status state) {
+                return true;
+            }
+
+            @Override
+            public boolean isRecurring() {
+                return false;
+            }
+
+            @Override
+            public int getTimeout() {
+                return 0;
+            }
+
+            @Override
+            public boolean processTimeout(long agentId, long seq) {
+                return true;
+            }
+        }, false, true, false);
+    }
+
+    @DB
+    private void initNuageNetworkOffering() {
+        Transaction.execute(new TransactionCallbackNoReturn() {
+            @Override
+            public void doInTransactionWithoutResult(TransactionStatus status) {
+                if (_networkOfferingDao.findByUniqueName(nuageVspSharedNetworkOfferingWithSGServiceName) == null) {
```

Comment:
Extract the call to ``_networkOfferDao`` to a local variable to improve readability.